wagtail/admin/views/collections.py

```diff
@@ -21,8 +21,8 @@ class Index(IndexView):
     header_icon = 'folder-open-1'
 
     def get_queryset(self):
-        # Only return children of the root node, so that the root is not editable
-        return Collection.get_first_root_node().get_children().order_by('name')
+        # Only return descendants of the root node, so that the root is not editable
+        return Collection.get_first_root_node().get_descendants().order_by('path')
```

Comment:
Changing this order will revert #4874. Unfortunately I can't see any way of ordering by name in a get_descendants query, other than enforcing that ordering on the `path` field via [treebeard's node_order_by setting](https://django-treebeard.readthedocs.io/en/latest/api.html) - and treebeard doesn't provide a non-destructive way to apply that retrospectively on an existing tree. (Doing it in a data migration would be fiddly, although not impossible, and the fact that we only have to do it on one level of the tree makes it slightly easier...)

Reply:
Well shoot. Working with the low level Treebeard stuff is outside my knowledgebase. Could you give me a hint on how one might go about making this data migration?

Reply:
I decided to bite the bullet and write the missing code for Treebeard :-) https://github.com/django-treebeard/django-treebeard/pull/165
I don't know how long it's going to take for that to make it into a new Treebeard release, but in the meantime we can copy that code into a helper function in Wagtail.

Reply:
Oh man, that would have *definitely* been way beyond my own skill level. 

There's a hefty amount of new code in that PR. How exactly would you suggest I port that into Wagtail as a "helper function"?

Reply:
Have now done this in #6190 - so we just need to add `node_order_by=['name']` to the Collection model now.

Reply:
So should `node_order_by=['name']` be added to Collection in this PR, or in a separate one?

Reply:
I stumbled upon #5888 and wonder if it is of any relevance to this

Reply:
@zerolab From a quick look, the current PR would make #5888 obsolete, since with nested collections the ordering logic proposed there (i.e. order by name, except for the root collection which goes at the top) would no longer be appropriate.

@coredumperror Yes, please add the `node_order_by` line to this PR - that way, this PR is functionally complete in its own right (and #6190 remains as an enhancement to assist people who need to migrate to the new ordering).